pkg/cloudprovider/cloudprovider.go

```diff
@@ -48,21 +70,48 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 }
 
 func (c *CloudProvider) List(ctx context.Context) ([]*karpv1.NodeClaim, error) {
-
-	// TODO: Implement this
-	return nil, nil
+	instances, err := c.instanceProvider.List(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("listing instances, %w", err)
+	}
+	var nodeClaims []*karpv1.NodeClaim
+	for i := range instances {
+		instanceType, err := c.resolveInstanceTypeFromInstance(ctx, instances[i])
+		if err != nil {
+			return nil, fmt.Errorf("resolving instance type, %w", err)
+		}
+		nc, err := c.resolveNodeClassFromInstance(ctx, instances[i])
+		if client.IgnoreNotFound(err) != nil {
+			return nil, fmt.Errorf("resolving nodeclass, %w", err)
+		}
+		nodeClaims = append(nodeClaims, c.instanceToNodeClaim(instances[i], instanceType, nc))
+	}
+	return nodeClaims, nil
 }
 
 func (c *CloudProvider) Get(ctx context.Context, providerID string) (*karpv1.NodeClaim, error) {
-
-	// TODO: Implement this
-	return nil, nil
+	id, err := utils.ParseInstanceID(providerID)
+	if err != nil {
+		return nil, fmt.Errorf("getting instance ID, %w", err)
```

Comment:
So, as we discussed, log the error in the closest location, excluding the tools function.